warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/handler/edge/ProtobufEdgeDataTypeHandler.java

```diff
@@ -634,45 +620,73 @@ public long process(KEYIN key, RawRecordContainer event, Multimap<String,Normali
                 continue;
             }
             
-            // If within the same group, then within each subgroup that are in common for both the sink and source
-            if (sourceGroup.equals(sinkGroup) && (sourceGroup != NO_GROUP)) {
-                Set<String> commonKeys = mSource.keySet();
-                commonKeys.retainAll(mSink.keySet());
-                /**
-                 *
-                 * We are using the intersection of 2 sets here to make sure we only loop over edges that we will create. Previously we would loop over all the
-                 * possible edges even if they were at different levels of nesting.
-                 *
-                 */
-                for (String subGroup : commonKeys) {
-                    for (NormalizedContentInterface ifaceSource : mSource.get(subGroup)) {
-                        for (NormalizedContentInterface ifaceSink : mSink.get(subGroup)) {
-                            EdgeDataBundle edgeValue = createEdge(edgeDef, event, ifaceSource, sourceGroup, subGroup, ifaceSink, sinkGroup, subGroup,
-                                            edgeAttribute2, edgeAttribute3, normalizedFields, depthFirstList, loadDateStr, activityDate, validActivityDate);
-                            if (edgeValue != null) {
-                                
-                                // have to write out the keys as the edge values are generated, so counters get updated
-                                // and the system doesn't timeout.
-                                edgesCreated += writeEdges(edgeValue, context, contextWriter, validActivityDate, activityEqualsEvent, event.getDate());
+            // leave the old logic alone
+            if (!edgeDef.hasJexlPrecondition() || !edgeDef.isGroupAware()
+                            || !(matchingGroups.containsKey(sourceGroup) || matchingGroups.containsKey(sinkGroup))) {
```

Comment:
Something like this may improve readability, but definitely not required
```
// @formatter:off
if ( !edgeDef.hasJexlPrecondition() || 
     !edgeDef.isGroupAware() || 
     !(matchingGroups.containsKey(sourceGroup) || matchingGroups.containsKey(sinkGroup))) {
// @formatter:on
```